### PR TITLE
fix: MODE SEL manual visual improvements

### LINKED ISSUE #4034: LDG ELEV on ECAM when MAN MODE is selected
**Mod Version**

  "built": "2021-03-19T18:50:06+00:00",
  "ref": "master",
  "sha": "fcd8f930460051f660651ef98dc588f27282c65d",
  "actor": "davidwalschots",
  "event_name": "manual"

**Describe the bug**

According to the FCOM, when the MODE SELECT pushbutton in the CABIN PRESS overhead panel is set to manual, the landing elevation information should disappear from the ECAM PRESS and CRUISE pages. SYS1/2 should also disappear as the pressurisation is set to manual. 

**To Reproduce**

1. Spawn in any state.
2. Set the MODE SELECT pushbutton to manual.
3. Observe the ECAM PRESS and (if flying) CRUISE pages.

**Expected behavior**

The landing information should not be visible on the ECAM when the cabin press MODE is set to manual.

**Actual behavior**

(with MAN MODE selected)
image

**References**

image
image

**Additional context**

I will open a PR for this myself soon unless someone is already working on it.

Discord username (if different from GitHub): Commander Ga#1548

### PR DESCRIPTION
**Pull request by mjuhe**
Fixes #4034 

## Summary of Changes

- When MOD SEL is set to MAN: 
          - Removed LDG ELEV info from ECAM Press
          - Removed SYS1/2 display on ECAM Press
          - Removed LDG ELEV info from ECAM Cruise
- When MOD SEL is cycled (set to MAN for at least 10s and then reset to auto) the active cabin pressure control system is switched (SYS1 <-> SYS2).

## Screenshots (if necessary)

Before:
image

After:
image
image

## References

image
image
image
image
image
image

## Additional context

My first pull request :)

Limitations: 
          - The system switch will only happen if the ECAM PRESS page is displayed. This should be better implemented in a future PR to improve the representation of the press system as a whole.
          - The dials and values on the PRESS page point to simvars that are not currently being set anywhere. This I believe was intentional in PR #1373 awaiting the wasm system.

Discord username (if different from GitHub): Commander Ga#1548

## Testing instructions

1. Spawn in any state
2. Set the MODE SEL pushbutton in the CABIN PRESS overhead panel to manual
3. Observe the CRUISE page and PRESS page of the ECAM and verify no LDG ELEV information shows
4. Set the MODE SEL back to auto, if it's within 10s of the first press, the SYS showing on the PRESS page should stay the same, if it's after 10s, it should switch to the other SYS.
5. Verify no other issues appear with any of the other ECAM pages in any stages of flight.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Benjozork** [comment]
Needs to be validated for EIS2

**Comment by mjuhe** [comment]
Anything I can do to push this forward? I don't have access to the EIS2 myself I'm afraid.

**Comment by pilotJesse** [comment]
Category: QA Trial
Name: Jessee #8759
Date of testing: 02/04/2021
Version of the sim: 0.5.4
PR Tested: #5416
Tier of testing: 1

Changes to observe:
When MOD SEL is set to MAN:
- Removed LDG ELEV info from ECAM Press
- Removed SYS1/2 display on ECAM Press
- Removed LDG ELEV info from ECAM Cruise
When MOD SEL is cycled (set to MAN for at least 10s and then reset to auto) the active cabin pressure control system is switched (SYS1 <-> SYS2).

Testing technique: Observed changes to PRESS sys page from MODE SEL MAN-> AUTO

Steps you've made to test the Pull Request:
1. Aircraft on stand, GPU powered on Batt 1/2.
2. CAB PRESS observed in AUTO then observed once MODE SEL > MAN.
3. CAB PRESS page observed all requirements: SYS1/2, LDG ELEV extinguished from page
2. MODE SEL MAN -> AUTO within 10 sec= SYS 1 retained 
3. MODE SEL MAN -> AUTO over 10 sec= SYS 1 -> SYS 2
4. Requirements met, test successful

Flight notes: N/A

Media:
MAN OFF
MAN ON
MAN OFF (AUTO)
SYS OFF


Issues (If Any): No issues (additional comments below)

Overall Rating: Good
Conclusions of the testing: Requirements met, additional test on the MAN V/S CTL switch does not open or close the outflow valve in order to change the cabin v/s and manipulate MAN PRES. Switch is though spring loaded in neutral position therefore correct position.

**Review by Benjozork — APPROVED** [review]
Validated EIS2 S13-2.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_PRESS.js [hunk 1]
```diff
@@ -294,9 +298,28 @@ var A320_Neo_LowerECAM_PRESS;
...
                 this.oldManModValue = 1;
             } else if (!manMode && this.oldManModValue) {
                 this.htmlMANtext.setAttribute("visibility", "hidden");
+                this.htmlLdgElevText.setAttribute("visibility", "visible");
+                this.htmlLdgElevValue.setAttribute("visibility", "visible");
+                this.htmlLdgElevTitle.setAttribute("visibility", "visible");
+                this.htmlLdgElevValueUnit.setAttribute("visibility", "visible");
+                // The switch from one system to the other should only happen if there's at least 10s between presses - placeholder until PRESS system is fully simulated
+                if ((Date.now() - this.manModeTime) > 10000) {
+                    activeSystem === 1 ? SimVar.SetSimVarValue("L:CPC_SYS1", "Bool", 0) : SimVar.SetSimVarValue("L:CPC_SYS2", "Bool", 1);
```
Comment (Benjozork) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_PRESS.js:317]: > `"L:CPC_SYS1"` ?

is this a new variable

Comment (mjuhe) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_PRESS.js:317]: No, it was already in the PRESS page code created in PR #1373 (line no. 220). I don't think it's being set or modified anywhere else at the moment though.

File: flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_CRZ.js [hunk 2]
```diff
@@ -181,6 +185,23 @@ var A320_Neo_LowerECAM_CRZ;
...
 
+            // Landing elevation
+            const manMode = SimVar.GetSimVarValue("L:A32NX_CAB_PRESS_MODE_MAN", "Bool");
+
+            if (manMode && !this.oldManMode) {
+                this.LandingElevationTitle.setAttribute("visibility", "hidden");
+                this.LandingElevationMode.setAttribute("visibility", "hidden");
+                this.LandingElevation.setAttribute("visibility", "hidden");
+                this.LandingElevationUnit.setAttribute("visibility", "hidden");
+                this.oldManMode = 1;
```
Comment (tracernz) [inline flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_CRZ.js:196]: Old man mode eh? 😆

**Review by tracernz — COMMENTED** [review]
I think another test after the rebase should be enough to merge this. 😀

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 7.04.2021
Version of the sim: 1.14.6.0
PR Tested: #4037
Tier of testing: 1
Changes to observe: MOD SEL on ECAM PRESS page
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned C&D at EDDP
2.) Start up AC and took off to CRZ alt
3.) Set CABIN PRESS MODE SEL to manual
- Removed LDG ELEV
- Removed SYS
- Shows MAN
- Removed LDG ELEV on Cruise page

4.) Set CABIN PRESS MODE SEL to auto (>10s manual)
- Shows LDG ELEV
- Shows SYS
- Removed LDG ELEV on Cruise page
- Switched to the other SYS

5.) Set CABIN PRESS MODE SEL to manual
- Removed LDG ELEV
- Removed SYS
- Shows MAN
- Removed LDG ELEV on Cruise page

6.) Set CABIN PRESS MODE SEL to auto (<10s manual)
- Shows LDG ELEV
- Shows SYS
- Removed LDG ELEV on Cruise page
- Stays at same SYS

Issues: -

Overall Rating: Good

Conclusions of the testing: Works

Hope this helps :-)

**Pull request merged**